code/game/gamemodes/miniantags/revenant/revenant_abilities.dm

```diff
@@ -259,30 +263,148 @@
 	cast_amount = 45
 	unlock_amount = 150
 	action_icon_state = "malfunction"
+	aoe_range = 2
 
-/obj/effect/proc_holder/spell/aoe_turf/revenant/malfunction/create_new_targeting()
-	var/datum/spell_targeting/aoe/turf/T = new()
-	T.range = 2
-	return T
+/obj/effect/proc_holder/spell/aoe/revenant/malfunction/create_new_targeting()
+	var/datum/spell_targeting/aoe/turf/targeting = new()
+	targeting.range = aoe_range
+	return targeting
 
 //A note to future coders: do not replace this with an EMP because it will wreck malf AIs and gang dominators and everyone will hate you.
-/obj/effect/proc_holder/spell/aoe_turf/revenant/malfunction/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+/obj/effect/proc_holder/spell/aoe/revenant/malfunction/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
 	if(attempt_cast(user))
 		for(var/turf/T in targets)
 			INVOKE_ASYNC(src, PROC_REF(effect), user, T)
 
-/obj/effect/proc_holder/spell/aoe_turf/revenant/malfunction/proc/effect(mob/living/simple_animal/revenant/user, turf/T)
+/obj/effect/proc_holder/spell/aoe/revenant/malfunction/proc/effect(mob/living/simple_animal/revenant/user, turf/T)
 	T.rev_malfunction(TRUE)
 	for(var/atom/A in T.contents)
 		A.rev_malfunction(TRUE)
 
+/**
+ * Makes objects be haunted and then throws them at conscious people to do damage, spooky!
+ */
+/obj/effect/proc_holder/spell/aoe/revenant/haunt_object
+	name = "Haunt Objects"
+	desc = "Empower nearby objects to you with ghostly energy, causing them to attack nearby mortals. \
+		Items closer to you are more likely to be haunted."
+	action_icon_state = "haunt"
+	base_cooldown = 60 SECONDS
+	unlock_amount = 150
+	cast_amount = 50
+	stun = 3 SECONDS
+	reveal = 10 SECONDS
+	aoe_range = 7
+	/// The maximum number of objects to haunt
+	var/max_targets = 7
+	/// A list of all attack timers started by this spell being cast
+	var/list/attack_timers = list()
+
+/obj/effect/proc_holder/spell/aoe/revenant/haunt_object/create_new_targeting()
+	var/datum/spell_targeting/aoe/targeting = new()
+	targeting.range = aoe_range
+	targeting.allowed_type = /obj/item
+	return targeting
+
+/obj/effect/proc_holder/spell/aoe/revenant/haunt_object/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(!attempt_cast(user))
+		return
+	var/successes = 0
+	for(var/obj/item/nearby_item as anything in targets)
+		// Don't throw around anchored things or dense things
+		// (Or things not on a turf but I am not sure if range can catch that)
+		if(nearby_item.anchored || nearby_item.density || nearby_item.move_resist == INFINITY || !isturf(nearby_item.loc))
+			continue
+		// Don't throw abstract things
+		if(nearby_item.flags & ABSTRACT)
+			continue
+		// Don't throw things we can't see
+		if(nearby_item.invisibility >= INVISIBILITY_REVENANT)
+			continue
+
+		var/distance_from_user = clamp(get_dist(get_turf(nearby_item), get_turf(user)), 1, 127) // get_dist() never returns more than 127, but same tile dists return -1
```

Comment:
wouldnt max() do the job then? no point in clamping between x and y if y is already the max.